presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -239,15 +256,20 @@ protected Node visitNode(Node node, Void context)
             return new Constraint<>(scanNode.get().getCurrentConstraint(), bindings -> true);
         }
 
-        private Map<ColumnHandle, String> getStatisticsColumnNames(TableStatistics statistics, ShowStats node, QualifiedName tableName)
+        private Map<ColumnHandle, String> getStatisticsColumnNames(TableStatistics statistics, TableHandle tableHandle)
         {
-            TableHandle tableHandle = getTableHandle(node, tableName);
-
             return statistics.getColumnStatistics()
                     .keySet().stream()
                     .collect(toMap(identity(), column -> metadata.getColumnMetadata(session, tableHandle, column).getName()));
         }
 
+        private Map<ColumnHandle, Type> getStatisticsColumnTypes(TableStatistics statistics, TableHandle tableHandle)
+        {
+            return statistics.getColumnStatistics()
+                    .keySet().stream()
+                    .collect(toMap(identity(), column -> metadata.getColumnMetadata(session, tableHandle, column).getType()));
```

Comment:
Maybe you could squash `getStatisticsColumnNames` and `getStatisticsColumnTypes` (i don't understand their names btw) and return just `Map<ColumnHandle, ColumnMetadata>`?